src/main/java/com/rarchives/ripme/ripper/rippers/DeviantartRipper.java

```diff
@@ -1,409 +1,531 @@
+ *         frequently as HTML source code...?
+ * 
+ * 
+ * 
+ *         Tested for:
+ * 
+ *         SFW:
+ * 
+ *         https://www.deviantart.com/apofiss/gallery/41388863/sceneries
+ *         https://www.deviantart.com/kageuri/gallery/
+ *         https://www.deviantart.com/kageuri/gallery/?catpath=/
+ *         https://www.deviantart.com/apofiss/favourites/39881418/gifts-and
+ *         https://www.deviantart.com/kageuri/favourites/
+ *         https://www.deviantart.com/kageuri/favourites/?catpath=/
+ * 
+ *         NSFW:
+ * 
+ *         https://www.deviantart.com/revpeng/gallery/67734353/Siren-Lee-Agent-of-S-I-R-E-N-S
+ * 
+ * 
+ * 
+ *         Login Data (PLEASE DONT ACTUALLY USE!!!):
+ * 
+ *         email: 5g5_8l4dii5lbbpc@byom.de
+ * 
+ *         username: 5g58l4dii5lbbpc
+ * 
+ *         password: 5g5_8l4dii5lbbpc
+ * 
+ * 
+ *
+ */
+public class DeviantartRipper extends AbstractHTMLRipper {
+
+	private final String username = "5g58l4dii5lbbpc";
+	private final String password = "5g5_8l4dii5lbbpc";
+	private int offset = 0;
+	private boolean usingCatPath = false;
+	private int downloadCount = 0;
+	private Map<String, String> cookies;
+	private DownloadThreadPool deviantartThreadPool = new DownloadThreadPool("deviantart");
+	private ArrayList<String> names = new ArrayList<String>();
+
+	// Constants
+	private final String referer = "https://www.deviantart.com/";
+	private final String userAgent = "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:65.0) Gecko/20100101 Firefox/65.0";
+
+	@Override
+	public DownloadThreadPool getThreadPool() {
+		return deviantartThreadPool;
+	}
+
+	public DeviantartRipper(URL url) throws IOException {
+		super(url);
+	}
+
+	@Override
+	protected String getDomain() {
+		return "deviantart.com";
+	}
+
+	@Override
+	public String getHost() {
+		return "deviantart";
+	}
+
+	@Override
+	protected Document getFirstPage() throws IOException {
+		login();
+		return Http.url(urlWithParams(this.offset)).cookies(getDACookie()).referrer(referer).userAgent(userAgent).get();
+	}
+
+	/**
+	 * Stores logged in Cookies. Needed for art pieces only visible to logged in
+	 * users.
+	 * 
+	 * 
+	 * @throws IOException when failed to load webpage or failed to read/write
+	 *                     cookies in file (used when running multiple instances of
+	 *                     RipMe)
+	 */
+	private void login() throws IOException {
+
+		File f = new File("DACookie.toDelete");
+		if (!f.exists()) {
+			f.createNewFile();
+			f.deleteOnExit();
+
+			// Load login page
+			Response res = Http.url("https://www.deviantart.com/users/login").connection().method(Method.GET)
+					.referrer(referer).userAgent(userAgent).execute();
+
+			// Find tokens
+			Document doc = res.parse();
+			Element form = doc.getElementById("login");
+			String token = form.select("input[name=\"validate_token\"]").first().attr("value");
+			String key = form.select("input[name=\"validate_key\"]").first().attr("value");
+			System.out.println(
```

Comment:
Please use LOGGER for logging and not System.out.println(). LOGGER logs both to the cli and the log file.

Also please avoid using dashes for formatting

Reply:
Using LOGGER now.